stock_valuation_layer_total_value/models/stock_valuation_layer.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class StockValuationLayer(models.Model):
+    _inherit = "stock.valuation.layer"
+
+    original_value = fields.Float(
+        string="Original value",
+        compute="_compute_original_layer_values",
+    )
+    adjusted_unit_price = fields.Float(
+        string="New value",
+        compute="_compute_original_layer_values",
+    )
+
+    def _compute_original_layer_values(self):
+        original_value = ad_unit_price = 0
+        for rec in self:
+            if len(rec.stock_valuation_layer_ids):
+                children_value = sum(rec.stock_valuation_layer_ids.mapped("value"))
+                original_value = rec.value / children_value
```

Comment:
A ZeroDivisionError seems possible here

Reply:
@AaronHForgeFlow there can be other reasons like inventory revaluations.

Reply:
Ok, but in that case we should we remove the dependency to stock_landed_cost. If seems to me it is only used for tests.

Reply:
@AaronHForgeFlow I removed the dependency to stock_landed_cost